suite-native/atoms/src/Button/Button.tsx

```diff
@@ -1,118 +1,196 @@
+        disabledBackgroundColor: 'gray200',
+        textColor: 'gray0',
+        disabledTextColor: 'gray500',
+    },
+    secondary: {
+        backgroundColor: 'green',
+        onPressColor: 'green900',
+        disabledBackgroundColor: 'gray200',
+        textColor: 'gray0',
+        disabledTextColor: 'gray500',
+    },
+    tertiary: {
+        backgroundColor: 'gray200',
+        onPressColor: 'gray300',
+        disabledBackgroundColor: 'gray200',
+        textColor: 'gray800',
+        disabledTextColor: 'gray500',
+    },
+    danger: {
+        backgroundColor: 'red',
+        onPressColor: 'red700',
+        disabledBackgroundColor: 'gray200',
+        textColor: 'gray0',
+        disabledTextColor: 'gray500',
+    },
+} as const satisfies Record<ButtonColorSchemeName, ButtonColorScheme>;
 
-        const buttonColorSchemeStyles: Record<ButtonColorScheme, NativeStyleObject> = {
-            primary: {
-                backgroundColor: utils.colors.green,
-            },
-            gray: {
-                backgroundColor: utils.colors.gray300,
+const textSizeToVariantMap = {
+    small: 'hint',
+    medium: 'body',
+    large: 'body',
+} as const satisfies Record<ButtonSize, TypographyStyle>;
+
+const iconStyle = prepareNativeStyle((utils, { position }: IconStyleProps) => ({
+    extend: [
+        {
+            condition: position === 'left',
+            style: {
+                marginRight: utils.spacings.small,
             },
-            red: {
-                backgroundColor: utils.transparentize(0.9, utils.colors.red),
+        },
+        {
+            condition: position === 'right',
+            style: {
+                marginLeft: utils.spacings.small,
             },
-        };
+        },
+    ],
+}));
 
+export const buttonStyle = prepareNativeStyle<ButtonStyleProps>(
+    (utils, { size, colorScheme, isDisabled, isPressed }) => {
+        const schemeColors = buttonSchemeToColorsMap[colorScheme];
         return {
             flexDirection: 'row',
             justifyContent: 'center',
             alignItems: 'center',
             borderRadius: utils.borders.radii.round,
-            ...buttonSizeStyles[size],
-            ...buttonColorSchemeStyles[colorScheme],
-            extend: {
-                condition: isDisabled,
-                style: {
-                    backgroundColor: utils.colors.gray200,
+            backgroundColor: utils.colors[schemeColors.backgroundColor],
+            extend: [
+                {
+                    condition: isDisabled,
+                    style: {
+                        backgroundColor: utils.colors[schemeColors.disabledBackgroundColor],
+                    },
                 },
-            },
+                {
+                    condition: isPressed,
+                    style: {
+                        backgroundColor: utils.colors[schemeColors.onPressColor],
+                    },
+                },
+                {
```

Comment:
Can you do same config object for sizes as you did for colors?

Reply:
fixed in 0f7794d